lxd/images.go

```diff
@@ -1164,6 +1221,34 @@ func autoUpdateImagesInProject(ctx context.Context, d *Daemon, project string) e
 			continue
 		}
 
+		// In theory, pools should contain at most 1 value. That is because the call to GetImage
+		// will fail if there are multiple matches.
+		// We intentionally ignore the error.
+		pools, _ := d.cluster.GetPoolsWithRemoteImage(fingerprint)
+
+		if len(nodes) > 1 && len(availableNodeIDs) > 0 && len(pools) == 1 {
+			isRemoteStorage, _ := d.cluster.IsRemoteStorage(pools[0])
```

Comment:
I'm confused, doesn't `GetPoolsWithRemoteImage` already do that? From what I remember of that function, you were specifically filtering on `node_id IS NULL` which would only ever be true of remote storage pools.

Reply:
You're right. This check is unnecessary.